pepys_import/core/store/common_db.py

```diff
@@ -107,30 +107,43 @@ def get_sensor(self, data_store, sensor_name=None, sensor_type=None, privacy=Non
 
 
 class DatafileMixin:
-    def create_state(self, data_store, sensor, timestamp, parser_name):
+    def create_state(self, data_store, platform, sensor, timestamp, parser_name):
         state = data_store.db_classes.State(
             sensor_id=sensor.sensor_id, time=timestamp, source_id=self.datafile_id
         )
+        state.platform_name = platform.name
+        state.sensor_name = sensor.name
         self.measurements[parser_name].append(state)
         return state
 
-    def create_contact(self, data_store, sensor, timestamp, parser_name):
+    def create_contact(self, data_store, platform, sensor, timestamp, parser_name):
         contact = data_store.db_classes.Contact(
             sensor_id=sensor.sensor_id, time=timestamp, source_id=self.datafile_id
         )
+        contact.platform_name = platform.name
+        contact.sensor_name = sensor.name
         self.measurements[parser_name].append(contact)
         return contact
 
     def create_comment(
-        self, data_store, platform_id, timestamp, comment, comment_type, parser_name
+        self,
+        data_store,
+        platform,
+        sensor,
```

Comment:
@BarisSari - Note: we don't store `sensor` for comments.  I guess you could use `None` or "N/A" instead.  I expect subsequent reporting logic relies on it being present.

Reply:
Sorry that I missed out. I removed the argument and replaced it with "N/A":

https://github.com/debrief/pepys-import/pull/196/commits/51b5afd15aa81352a632f60e3300a47fcc6a7655